pkg/controllers/provisioning/v1alpha1/allocation/filter.go

```diff
@@ -46,11 +46,11 @@ func (f *Filter) GetProvisionablePods(ctx context.Context, provisioner *v1alpha1
 	provisionable := []*v1.Pod{}
 	for _, pod := range pods.Items {
 		if err := functional.ValidateAll(
-			func() error { return f.isUnschedulable(&pod) },
-			func() error { return f.matchesProvisioner(&pod, provisioner) },
-			func() error { return f.hasSupportedSchedulingConstraints(&pod) },
-			func() error { return f.toleratesTaints(&pod, provisioner) },
-			func() error { return f.hasSupportedLabels(&pod) },
+			func() error { return f.isUnschedulable(ctx, &pod) },
+			func() error { return f.matchesProvisioner(ctx, &pod, provisioner) },
+			func() error { return f.hasSupportedSchedulingConstraints(ctx, &pod) },
+			func() error { return f.toleratesTaints(ctx, &pod, provisioner) },
+			func() error { return f.withValidConstraints(ctx, &pod, provisioner) },
```

Comment:
Maybe I'm missing something, but it looks like none of these 5 validation functions and their sub-functions use contexts. Do we need these contexts here?

Reply:
Knative has a logging construct (logger.FromContext(ctx).Info ...) and other amazing utilities around the ctx object (i.e. reading global configuration values from a configmap). 

I had left this in as a best practice for passing contexts around. However, I think this is probably premature, and easy to add as necessary in the future, so I'll rip this out for now. Thanks!